tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaHelper.cs

```diff
@@ -94,6 +95,54 @@ internal static class KafkaHelper
             return scope;
         }
 
+        private static long TryGetSize(object obj)
+        {
+            if (obj == null)
+            {
+                return 0;
+            }
+
+            if (obj is byte[] bytes)
+            {
+                return bytes.Length;
+            }
+
+            if (obj is string str)
+            {
+                return Encoding.UTF8.GetByteCount(str);
+            }
+
+            return 0;
+        }
+
+        private static long GetMessageSize(IMessage message)
```

Comment:
This is a perf optimisation due to the way duck typing works, but you _generally_ should avoid referencing duck types in methods directly. Instead, use a generic and condition

```suggestion
        private static long GetMessageSize<T>(T message)
            where T: IMessage
```